### PR TITLE
sclang: fix rounding in SimpleNumber -asTimeString

### LINKED ISSUE #4660: .asTimeString rounding error
## Environment

* SuperCollider version: 3.10.3
* Operating system: macOS 10.13
* Other details (Qt version, audio driver, etc.):

## Steps to reproduce

```supercollider
0.999.asTimeString; //OK
//-> 00:00:00.999

0.9999.asTimeString; //msecs rounded up
//-> 00:00:00.1000

0.99.asTimeString(precision: 0.1); //msecs rounded up
//-> 00:00:00.1000
```

## Expected vs. actual behavior

When the fraction of a second is rounded up (based on the `precision` parameter in `.asTimeString`), the fractional part of the string is being rounded up by itself and this is not propagated to the integer portion of the string.

### PR DESCRIPTION
**Pull request by dyfer**
## Purpose and Motivation

Fixes #4660

## Types of changes

- Bug fix

## To-do list

- [x] Code is tested
- [x] All tests are passing
- [x] This PR is ready for review
- [x] rebase to cleanup commits (after review)

### CODE REVIEW

File: SCClassLibrary/Common/Math/SimpleNumber.sc [hunk 1]
```diff
@@ -709,8 +711,7 @@ SimpleNumber : Number {
 
 		// min value of precision is 0.001; this ensures that we stick to 3 decimal places in the
```
Comment (snappizz) [inline SCClassLibrary/Common/Math/SimpleNumber.sc]: this comment is in the wrong place now. should be above `precision = max(precision, 0.001);` right?

File: testsuite/classlibrary/TestSimpleNumber.sc [hunk 2]
```diff
@@ -54,6 +54,12 @@ TestSimpleNumber : UnitTest {
 		this.assert(actual == expected, "expected %, got %".format(expected, actual));
 	}
 
+	test_asTimeString_millisecondPrecisionRounding {
+		var expected = "00:00:01.000";
+		var actual = 0.9999.asTimeString(precision: 0.001);
+		this.assert(actual == expected, "expected %, got %".format(expected, actual));
```
Comment (snappizz) [inline testsuite/classlibrary/TestSimpleNumber.sc]: this should be `assertEquals` and have a more descriptive comment. (feel free to refactor the surrounding code to use `assertEquals` also)

Comment (dyfer) [reply testsuite/classlibrary/TestSimpleNumber.sc]: Sure.
- should I leave comments pretty much as they are?
- should I set `report` parameter of `assertEquals` to `true` (default)?

Comment (snappizz) [reply testsuite/classlibrary/TestSimpleNumber.sc]: yeah leave it at default

**Comment by dyfer** [comment]
@snappizz I changed all surrounding comments (tests for `.asTimeString`). Before I remove the old ones which are now commented out (and rebase) what do you think about them?

(also I don't know what's the CI problem about...)

**Comment by snappizz** [comment]
great, thanks! just remove the commented-out lines and we'll be good to go

**Pull request merged**